help_to_heat/frontdoor/eligibility.py

```diff
@@ -30,25 +28,34 @@ def calculate_eligibility(session_data):
     country = session_data.get("country")
     benefits = session_data.get("benefits")
     property_type = session_data.get("property_type")
+    own_property = session_data.get("own_property")
 
     # "Scenario 0"
     if property_type == "Park home":
         return ("GBIS",)
-
-    # Scenario 1
+    
+    # ECO4 and GBIS scenario 1 - home owner
     if country in country_council_tax_bands:
-        if council_tax_band in country_council_tax_bands[country]["eligible"]:
-            if epc_rating in ("E", "F", "G", "Not found"):
+        if own_property in ("Yes, I own my property and live in it",):
+            if epc_rating in ("D","E", "F", "G", "Not found"):
                 if benefits in ("Yes",):
-                    logger.error("Scenario 1")
+                    logger.error("ECO4 scenario 1")
                     return ("GBIS", "ECO4")
-
-    # Scenario 2
+                
+    # ECO4 and GBIS scenario 2 - private rented (tenant or landlord)
     if country in country_council_tax_bands:
-        if council_tax_band in country_council_tax_bands[country]["ineligible"]:
+        if own_property in ("No, I am a tenant", "Yes, I am the property owner but I lease the property to one or more tenants"):
             if epc_rating in ("E", "F", "G", "Not found"):
                 if benefits in ("Yes",):
-                    logger.error("Scenario 2")
+                    logger.error("ECO4 scenario 2")
+                    return ("GBIS", "ECO4")
+                
+    # ECO4 and GBIS scenario 3 - social housing tenant
+    if country in country_council_tax_bands:
+        if own_property in ("No, I am a social housing tenant",):
+            if epc_rating in ("D", "E", "F", "G", "Not found"):
```

Comment:
I think scenarios 1 and 3 could be combined into a single scenario as the requirements in terms of EPC and benefits are the same, it would just be a case of combining the logic in lines 55 and 39? I've laid it out into 3 scenarios here as that's how Ian communicated the ECO4 eligibility to us.

Reply:
In this one instance, I'd be inclined to keep the logic matching our source of logic. Since the eligibility is a bit knotty (and potentially liable to change), I'm keen to have as little translation between requirements and code as possible.

Reply:
Sounds good, thanks!